src/AutoRest.CSharp/Mgmt/Output/MgmtObjectType.cs

```diff
@@ -80,7 +81,7 @@ protected override IEnumerable<ObjectTypeProperty> BuildProperties()
 
         private static bool IsSinglePropertyObject(ObjectTypeProperty property)
         {
-            if (!property.Declaration.Type.TryCast<ObjectType>(out var objType))
+            if (property.Declaration.Type is not { IsFrameworkType: false, Implementation: ObjectType objType })
```

Comment:
There are a lot of check statement like `is { IsFrameowkrType: false, Implementation: {type}}`. Can we provide a `IsCustomType{type}` method to replace `TryCast`?

Reply:
that is the TryCast with a new name. The purpose of this change is to remove it because we are using two different approaches of getting it, one is `TryCast` the other way is this is pattern matching. And currently `is` way has a overwhelming number therefore I am removing the method to get consistency.

Reply:
If it's a common pattern, I'd prefer a utility method with better name, which provides better readability.

Reply:
well then I could revert this change.
I remove this for alignment because there are two different usage: call TryCast or use `is`
`is` is more widely used, therefore I would like to remove this.